econml/iv/dr/_dr.py

```diff
@@ -315,6 +315,7 @@ def __init__(self, *,
         self.opt_reweighted = opt_reweighted
         super().__init__(discrete_instrument=discrete_instrument,
                          discrete_treatment=discrete_treatment,
+                         treatment_featurizer=None,
```

Comment:
I believe we have been avoiding enabling treatment featurization for estimators where treatments are required to be either discrete or 1d/binary.

There are some IV methods that take in any type of treatment. We may be able to enable treatment featurization for those.

Reply:
Yeah, I think _BaseDRIV can take a continuous treatment (otherwise, discrete_treatment should not be an init arg), which is why I asked.  I think we should enable treatment featurization in any class that allows continuous treatments, unless there's some other reason not to.

Having said that, it seems like DML is by far the most important use case to start with, so if it's a big hassle you can defer it and add an issue to revisit later.